Client/Frontend/Browser/EnhancedTrackingProtection/EnhancedTrackingProtectionVC.swift

```diff
@@ -0,0 +1,453 @@
+        button.layer.cornerRadius = 0.5 * ETPMenuUX.UX.closeButtonSize
+        button.clipsToBounds = true
+        button.setImage(UIImage(named: "close-medium"), for: .normal)
+        button.imageView?.contentMode = .scaleAspectFit
+    }
+
+    private let horizontalLine: UIView = .build { line in
+        line.backgroundColor = UIColor.theme.etpMenu.horizontalLine
+    }
+
+    // Connection Info view
+    private let connectionView = ETPSectionView(frame: .zero)
+
+    private let connectionImage: UIImageView = .build { image in
+        image.contentMode = .scaleAspectFit
+        image.clipsToBounds = true
+        image.layer.masksToBounds = true
+        image.layer.cornerRadius = 5
+    }
+
+    private let connectionLabel: UILabel = .build { label in
+        label.font = ETPMenuUX.Fonts.viewTitleLabels
+        label.numberOfLines = 0
+    }
+
+    private let connectionDetailArrow: UIImageView = .build { image in
+        image.image = UIImage(imageLiteralResourceName: "goBack").withRenderingMode(.alwaysTemplate)
+        image.transform = CGAffineTransform(rotationAngle: .pi)
+    }
+
+    private let connectionButton: UIButton = .build { button in }
+
+    // TrackingProtection toggle View
+    private let toggleContainer: UIView = .build { view in
+        view.backgroundColor = .clear
+    }
+
+    private let toggleView: UIView = ETPSectionView(frame: .zero)
+
+    private let toggleLabel: UILabel = .build { label in
+        label.font = ETPMenuUX.Fonts.viewTitleLabels
+        label.numberOfLines = 0
+    }
+
+    private let toggleSwitch: UISwitch = .build { toggleSwitch in
+        toggleSwitch.isEnabled = true
+        toggleSwitch.onTintColor = .systemBlue
+    }
+
+    private let toggleStatusLabel: UILabel = .build { label in
+        label.font = ETPMenuUX.Fonts.detailsLabel
+        label.numberOfLines = 0
+    }
+
+    // Protection setting view
+    private let protectionView: UIView = ETPSectionView(frame: .zero)
+
+    private var protectionButton: UIButton = .build { button in
+        button.setTitle(Strings.TPProtectionSettings, for: .normal)
+        button.titleLabel?.font = ETPMenuUX.Fonts.viewTitleLabels
+        button.setTitleColor(.systemBlue, for: .normal)
+        button.contentHorizontalAlignment = .left
+    }
+
+    private var constraints = [NSLayoutConstraint]()
+
+    // MARK: - Variables
+
+    private var viewModel: EnhancedTrackingProtectionMenuVM
+    private var hasSetPointOrigin = false
+    private var pointOrigin: CGPoint?
+    var asPopover: Bool = false
+
+    private var toggleContainerShouldBeHidden: Bool {
+        return !viewModel.globalETPIsEnabled
+    }
+
+    private var protectionViewTopConstraint: NSLayoutConstraint?
+
+    // MARK: - View lifecycle
+
+    init(viewModel: EnhancedTrackingProtectionMenuVM) {
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
```

Comment:
What's the purpose of `@available(*, unavailable)`?

Reply:
Just being explicit about making sure no-one accidentally uses it. It's frivolous, in a way, but safer.